Tests/FluentAssertions.Specs/Primitives/DateOnlyAssertionSpecs.cs

```diff
@@ -0,0 +1,863 @@
+using System;
+using Xunit;
+using Xunit.Sdk;
+
+#if NET6_0_OR_GREATER
+
+namespace FluentAssertions.Specs.Primitives
+{
+    public class DateOnlyAssertionSpecs
+    {
+        #region (Not) Be
+
+        [Fact]
+        public void Should_succeed_when_asserting_dateonly_value_is_equal_to_the_same_value()
+        {
+            // Arrange
+            DateOnly dateOnly = new(2016, 06, 04);
```

Comment:
🤔 Looking at this, I'm wondering if we should change the month-based extensions to return `DateOnly` for .NET 6. Because normally you can write that date as `4.April(2016)`.

Reply:
Would be interesting. And if someone wants a datetime as return type, they would have to add time explicitly? 
Something like `4.April(2016).AsDateTime()` or `4.April(2016).WithTime()`?

Reply:
I like the idea.
If you wanted a `DateTime` instead of `DateOnly` you could write `4.April(2016).At(00, 00)`.

A problem here is that if we change the return type of `April(int year)` from `DateTime` to `DateOnly` this example breaks if upgrade FA and then later upgrade to net6.0
```cs
DateTime date = 4.April(2016);
```
with `CS0029	Cannot implicitly convert type 'System.DateOnly' to 'System.DateTime'`

Reply:
True. I thought that you just guessed with that error message, but it's the real deal ;-)

Kind of disappointed. It would make total sense to have implicit conversion from `DateOnly` to `DateTime`.

Reply:
There's an issue about it https://github.com/dotnet/runtime/issues/58734